File: core/clientLaya/game/src/shine/net/httpRequest/SimpleHttpRequest.ts
namespace Shine
{
	export class SimpleHttpRequest extends BaseHttpRequest 
	{
	    private _completeFunc: Function;
	
	    private _errorFunc: Function;
	
	    private _postData: string;
	
	    private _resultData: string;
	
	    constructor() 
	    {
			super();
	        this._method = HttpMethodType.Get;
	    }
	
	    protected write(): void
	    {
	        this._postStream = new BytesWriteStream();
	        this._postStream.writeUTFBytes(this._postData);
	    }
	
	    protected read(): void 
	    {
	        this._resultData = this._resultStream.readUFTBytes(this._resultStream.bytesAvailable());
	    }
	
	    protected onComplete(): void 
	    {
	        if (this._completeFunc != null)
	            this._completeFunc(this._resultData);
	    }
	
	    protected onError(): void
	    {
	        if (this._errorFunc != null)
	            this._errorFunc();
	    }
	
	    public static create(): SimpleHttpRequest 
	    {
	        return new SimpleHttpRequest();
	    }
	
	    /* http请求 */
	    public static httpRequest(url: string, method: number = HttpMethodType.Get, data: string = null, completeFunc: Function = null, errorFunc: Function = null): void 
	    {
	        if (url.indexOf('?') == -1) 
	        {
	            url += "?v=" + MathUtils.randomInt(1000000);
	        }
	        else 
	        {
	            url += "&v=" + MathUtils.randomInt(1000000);
	        }
	            
	        var request: SimpleHttpRequest = SimpleHttpRequest.create();
	        request._url = url;
	        request._method = method;
	        request._postData = data;
	        request._completeFunc = completeFunc;
	        request._errorFunc = errorFunc;
	
	        request.send();
	    }
	}
}

File: core/client/game/src/shine/net/httpRequest/SimpleHttpRequest.cs
using System;

namespace ShineEngine
{
	/// <summary>
	/// 简版http请求
	/// </summary>
	public class SimpleHttpRequest:BaseHttpRequest
	{
		private Action<string> _completeFunc;

		private Action _errorFunc;

		private string _postData;

		private string _resultData;

		private SimpleHttpRequest()
		{
			_method=HttpMethodType.Get;
		}

		protected override void write()
		{
			_postStream=new BytesWriteStream();
			_postStream.writeUTFBytes(_postData);
		}

		protected override void read()
		{
			_resultData=_resultStream.readUTFBytes(_resultStream.bytesAvailable());
		}

		protected override void onComplete()
		{
			if(_completeFunc!=null)
				_completeFunc(_resultData);
		}

		protected override void onError()
		{
			if(_errorFunc!=null)
				_errorFunc();
		}

		public static SimpleHttpRequest create()
		{
			return new SimpleHttpRequest();
		}

		/// <summary>
		/// http请求
		/// </summary>
		public static void httpRequest(string url,Action<string> completeFunc)
		{
			httpRequest(url,HttpMethodType.Get,null,completeFunc);
		}

		/// <summary>
		/// http请求
		/// </summary>
		public static void httpRequest(string url,int method=HttpMethodType.Get,string data=null,Action<string> completeFunc=null,Action errorFunc=null)
		{
			if(url.IndexOf('?')==-1)
			{
				url+="?v=" + MathUtils.randomInt(1000000);
			}
			else
			{
				url+="&v=" + MathUtils.randomInt(1000000);
			}

			SimpleHttpRequest request=SimpleHttpRequest.create();
			request._url=url;
			request._method=method;
			request._postData=data;
			request._completeFunc=completeFunc;
			request._errorFunc=errorFunc;

			request.send();
		}
	}
}